fdbserver/TLogServer.actor.cpp

```diff
@@ -3599,14 +3656,16 @@ ACTOR Future<Void> tLog(IKeyValueStore* persistentData,
 			// Disk errors need a chance to kill this actor.
 			wait(delay(0.000001));
 
-			if (recovered.canBeSet())
+			if (recovered.canBeSet()) {
 				recovered.send(Void());
+			}
 
 			if (!self.durableClusterId.isValid()) {
 				self.sharedActors.send(updateDurableClusterID(&self));
 			}
 			self.sharedActors.send(commitQueue(&self));
 			self.sharedActors.send(updateStorageLoop(&self));
+			self.sharedActors.send(checkUpdateEncryptionAtRestMode(&self));
```

Comment:
Can we do this only when `SERVER_KNOBS->ENABLE_TLOG_ENCRYPTION` is on?

Reply:
Intention of the routine is two folds:
1. Update `persistentState` encryption status if not enabled - new TLog
2. Ensure the state matches with cluster configuration - restore TLog

Guarding the actor under KNOB would defeat #2 incase cluster config got updated in runtime (Encryption state can't be toggled). Once again, the KNOB is a stop-gap solution and soon will be replaced with already existing `db-config` which can be set only during cluster creation (sticky bit).